internal/datacoord/import_checker.go

```diff
@@ -210,6 +210,16 @@ func (c *importChecker) checkPreImportingJob(job ImportJob) {
 		return
 	}
 
+	err := CheckDiskQuota(job, c.meta, c.imeta)
+	if err != nil {
+		log.Warn("import failed, disk quota exceeded", zap.Int64("jobID", job.GetJobID()), zap.Error(err))
+		err = c.imeta.UpdateJob(job.GetJobID(), UpdateJobState(internalpb.ImportJobState_Failed), UpdateJobReason(err.Error()))
```

Comment:
I suggest refraining from immediately throwing an error. Instead, it could remain in an "InProgress" state, but with the reason clearly stated as insufficient quota.